client-spark/common/src/main/java/org/apache/uniffle/shuffle/BlockIdSelfManagedShuffleWriteClient.java

```diff
@@ -77,6 +77,31 @@ public void reportShuffleResult(
     managerClientSupplier.get().reportShuffleResult(request);
   }
 
+  @Override
+  public void reportShuffleResult(
+      Map<ShuffleServerInfo, Map<Integer, Set<Long>>> serverToPartitionToBlockIds,
+      String appId,
+      int shuffleId,
+      long taskAttemptId,
+      int bitmapNum,
+      Set<ShuffleServerInfo> reportFailureServers,
+      boolean enableWriteFailureRetry) {
+    Map<Integer, List<Long>> partitionToBlockIds = new HashMap<>();
+    for (Map<Integer, Set<Long>> k : serverToPartitionToBlockIds.values()) {
+      for (Map.Entry<Integer, Set<Long>> entry : k.entrySet()) {
+        int partitionId = entry.getKey();
+        partitionToBlockIds
+            .computeIfAbsent(partitionId, x -> new ArrayList<>())
+            .addAll(entry.getValue());
+      }
+    }
+
+    RssReportShuffleResultRequest request =
+        new RssReportShuffleResultRequest(
+            appId, shuffleId, taskAttemptId, partitionToBlockIds, bitmapNum);
+    managerClientSupplier.get().reportShuffleResult(request);
```

Comment:
It should also support the stage attempt id as the condition for blockId selfManaged of spark, and then support deleting when stage retry.

Reply:
Maybe this could be supported in the another PR, if you dont want to do it, I'm willing to submit  PR

Reply:
> Maybe this could be supported in the another PR, if you dont want to do it, I'm willing to submit PR

I'll do that again, but the next PR is done.